services/vehicle/src/main/java/com/wedriveu/services/vehicle/boundary/nearest/VehicleElectionVerticle.java

```diff
@@ -0,0 +1,84 @@
+package com.wedriveu.services.vehicle.boundary.nearest;
+
+import com.wedriveu.services.shared.rabbitmq.RabbitMQConfig;
+import com.wedriveu.services.shared.utilities.Constants;
+import com.wedriveu.services.shared.utilities.Log;
+import com.wedriveu.services.vehicle.rabbitmq.Messages;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+import io.vertx.rabbitmq.RabbitMQClient;
+
+import static com.wedriveu.services.shared.utilities.Constants.BODY;
+import static com.wedriveu.services.shared.utilities.Constants.USERNAME;
+
+/**
+ * This Verticle uses RabbitMQ Vertx.x library to publish the chosen vehicle to the client.
+ *
+ * @author Marco Baldassarri
+ * @since 4/08/2017
+ */
+public class VehicleElectionVerticle extends AbstractVerticle {
```

Comment:
Good idea, I made it, but the `VehicleFinderVerticle` publisher has to remain quite the same for now as it has slightly different method implementation, different from the base ones